vermouth/processors/apply_rubber_band.py

```diff
@@ -227,13 +309,13 @@ def apply_rubber_band(molecule, selector,
     constants = compute_force_constants(distance_matrix, lower_bound,
                                         upper_bound, decay_factor, decay_power,
                                         base_constant, minimum_force)
-    connectivity = build_connectivity_matrix(molecule, res_min_dist - 1,
-                                             selection=selection)
-    # Set the force constant to 0 for pairs that are connected. `connectivity`
-    # is a matrix of booleans that is True when a pair is connected. Because
-    # booleans acts as 0 or 1 in operation, we multiply the force constant
-    # matrix by the oposite (OR) of the connectivity matrix.
-    constants *= ~connectivity
+    connected = build_connectivity_matrix(molecule, res_min_dist - 1,
+                                          selection=selection)
+    same_domain = build_pair_matrix(molecule, domain_criterion,
+                                    selection=selection)
+    can_be_linked = (~connected) & same_domain
```

Comment:
This only works because you enumerate over `graph.nodes` in the same order in both `build_connectivity_matrix` (correspondence) and `build_pair_matrix`. Fragile to say the least. Especially if 2 months from now we decide it's more efficient to iterate over `selection` in `build_pair_matrix` rather than over `graph.nodes`.
At the very least add a comment to those bits of code, but ideally, construct a correspondence dict elsewhere (e.g. here).

Reply:
Why don't we build the selection outside these two functions. We kind of do the work double here. Just have a function `selection_to_nodes(graph,  selection)` which returns the collection of nodes that is given to the two functions. In that way it will always be in the same order, without having to build any correspondence dict.

Reply:
@jbarnoud why do we actually build selections for these functions? I just tested and it works if I simply take the one constructed here 

https://github.com/marrink-lab/vermouth-martinize/blob/d3a8342bf657d1664b87e814cb807e1e76035334/vermouth/processors/apply_rubber_band.py#L307-L315

Reply:
IIRC one selection is there to select the backbone (or whatever you want to apply the rubber band to), and one is there to see if it is the same chain.